File: Java/Logic1/In1To10.java
/**
 * CodingBat Java Activity "in1To10" from Logic1.
 * codingbat.com
 */
public class In1To10
{
	private static int failures = 0;
	private static final String METHOD_NAME = "in1To10";

	/**
	 * INSTRUCTIONS:
	 * Given a number n, return true if n is in the range 1..10, inclusive.
	 * Unless outsideMode is true, in which case return true if the number
	 * is less or equal to 1, or greater or equal to 10.
	 * 
	 * in1To10(5, false) -> true
	 * in1To10(11, false) -> false
	 * in1To10(11, true) -> true
	 * 
	 */
	public static boolean in1To10(int n, boolean outsideMode)
	{
		// ADD YOUR CODE HERE
	}

	public static void main(String[] args)
	{
		testCase(5, false, true);
		testCase(11, false, false);
		testCase(11, true, true);
		testCase(10, false, true);
		testCase(10, true, true);
		testCase(9, false, true);
		testCase(9, true, false);
		testCase(1, false, true);
		testCase(1, true, true);
		testCase(0, false, false);
		testCase(0, true, true);
		testCase(-1, false, false);
		testCase(-1, true, true);
		testCase(99, false, false);
		testCase(99, true, true);
		
		System.out.println();
		System.out.println(failures + " failed test cases.");
	}

	private static void testCase(int a, boolean b, boolean result)
	{
		String r;
		try
		{
			r = ""+in1To10(a, b);
			if (r.equals(""+result))
			{
				System.out.print("PASS: ");
			}
			else
			{
				System.out.print("FAIL: ");
				failures ++;
			}
		}
		catch (Exception e)
		{
			r = e.getMessage();
			System.out.println("FAIL: ");
			failures ++;
		}
		System.out.println(METHOD_NAME+"("+a+") -> "+r + ", EXPECTED: "+result);
	}

}

File: Python/logic-1/in1to10.py
'''
CodingBat Python Activity "in1to10" from logic-1.
codingbat.com
'''

failures = 0

def in1to10(n, outside_mode):
    '''
Given a number n, return True if n is in the range 1..10, inclusive.
Unless outside_mode is True, in which case return True if the number is less or equal to 1,
or greater or equal to 10.


in1to10(5, False) → True
in1to10(11, False) → False
in1to10(11, True) → True
'''
    pass #Replace pass with your code to implement this function





def catchFailures(testCase, result):
    global failures
    
    try:
        assert eval(testCase) == result, "Test Case Failed: "+testCase+ " → "+str(result)
    except Exception as e:
        failures += 1
        print(e)
            


def test():
    global failures
    
    catchFailures("in1to10(5, False)", True)
    catchFailures("in1to10(11, False)", False)
    catchFailures("in1to10(11, True)", True)
    catchFailures("in1to10(10, False)", True)
    catchFailures("in1to10(10, True)", True)
    catchFailures("in1to10(9, False)", True)
    catchFailures("in1to10(9, True)", False)
    catchFailures("in1to10(1, False)", True)
    catchFailures("in1to10(1, True)", True)
    catchFailures("in1to10(0, False)", False)
    catchFailures("in1to10(0, True)", True)
    catchFailures("in1to10(-1, False)", False)
    catchFailures("in1to10(-1, True)", True)
    catchFailures("in1to10(99, False)", False)
    catchFailures("in1to10(-99, True)", True)

    if failures > 0:
        print("\r\n "+ str(failures) + " failed test case(s)   :(")
    else:
        print("Good job -- problem solved!")


test()
